sources/packages/api/src/services/application/application.service.ts

```diff
@@ -638,4 +646,118 @@ export class ApplicationService extends RecordDataModelService<Application> {
       });
     return query.getOne();
   }
+
+  /**
+   * Creates a new Student Application overriding the current one
+   * in order to rollback the process and start the assessment
+   * all over again.
+   * @param locationId location id executing the COE rollback.
+   * @param applicationId application to be rolled back.
+   * @returns the overridden and the newly created application.
+   */
+  async overrideApplicationForCOE(
+    locationId: number,
+    applicationId: number,
+  ): Promise<ApplicationOverriddenResult> {
+    const appToOverride = await this.repo.findOne({
+      id: applicationId,
+      location: { id: locationId },
+    });
+
+    if (!appToOverride) {
+      throw new CustomNamedError(
+        "Student Application not found or the location does not have access to it",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (
+      appToOverride.applicationStatus !== ApplicationStatus.enrollment ||
+      appToOverride.coeStatus !== COEStatus.required
+    ) {
+      throw new CustomNamedError(
+        `Student Application is not in the expected status. The application must be in application status '${ApplicationStatus.enrollment}' and COE status '${COEStatus.required}' to be override.`,
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
+
+    // Change status of the current application being overwritten.
+    appToOverride.applicationStatus = ApplicationStatus.overwritten;
+    appToOverride.applicationStatusUpdatedOn = getUTCNow();
+
+    // Create a new application based in the previous one.
+    const newApplication = new Application();
+    newApplication.data = appToOverride.data;
+    newApplication.applicationNumber = appToOverride.applicationNumber;
+    newApplication.student = {
+      id: appToOverride.studentId,
+    } as Student;
+    newApplication.location = {
+      id: appToOverride.locationId,
+    } as InstitutionLocation;
+    newApplication.pirProgram = {
+      id: appToOverride.pirProgramId,
+    } as EducationProgram;
+    newApplication.programYear = {
+      id: appToOverride.programYearId,
+    } as ProgramYear;
+    newApplication.offering = {
+      id: appToOverride.offeringId,
+    } as EducationProgramOffering;
+    newApplication.pirStatus = ProgramInfoStatus.required;
+    newApplication.applicationStatus = ApplicationStatus.submitted;
+    newApplication.applicationStatusUpdatedOn = getUTCNow();
+    newApplication.studentFiles = appToOverride.studentFilesIds.map(
+      (applicationFileId: number) =>
+        ({
+          id: applicationFileId,
+        } as ApplicationStudentFile),
+    );
+
+    await this.repo.save([appToOverride, newApplication]);
+
+    return {
+      overriddenApplication: appToOverride,
+      createdApplication: newApplication,
+    };
+  }
+
+  /**
+   * Starts an application assessment workflow associating
```

Comment:
Question, is this specific for a rollback application or generic? If specific could be please review comments? Happy to discuss

Reply:
This is being used during application submission and during COE Edit (this PR), that the one described in the PR description as below. Please let me know if further comments are needed.
"Create a centralized method to start the application workflow and assign the workflow id to the application. This method is being shared between the submit application method (already existent) and the new one added to COE;'

Reply:
Nope I think that's good then. thank you for claifying